t/mojo/prefork.t

```diff
@@ -133,4 +223,39 @@ ok !$graceful, 'server has been stopped immediately';
 is $tx->res->code, 200,          'right status';
 is $tx->res->body, 'works too!', 'right content';
 
+my $events = Mojo::UserAgent->new->get("$durl/events")->res->body;
+my $href   = decode_json($events);
+is ref $href, 'HASH', 'right response type';
+is ref $href->{events},  'HASH', 'right events key';
+is ref $href->{spawned}, 'HASH', 'right spawned key';
+is ref $href->{server},  'HASH', 'right server key';
+
+my $event_count   = scalar keys %{$href->{events}};
+my $spawned_count = scalar keys %{$href->{spawned}};
+my $server_count  = scalar keys %{$href->{server}};
+
+is $event_count,   2, 'right amount of events';
+is $spawned_count, 5, 'right amount of spawned servers';
+is $server_count,  1, 'right amount of master servers';
+
+sub all_values_one {
+  my ($href) = @_;
+  foreach my $val (values %$href) {
+    return 0 unless 1 == $val;
+  }
+  return 1;
+}
+
+is all_values_one($href->{events}),  1, '1 before_app_start trigger per process';
+is all_values_one($href->{spawned}), 1, '1 spawn trigger per process';
+
+my $foreign = 0;
+foreach my $pid (keys %{$href->{events}}) {
+  $foreign++ unless exists $href->{spawned}->{$pid} || exists $href->{server}->{$pid};
+}
+
+is $foreign, 0, 'no before_app_start ran in foreign processes';
+
+kill 'TERM', $dpid;
```

Comment:
This is a mess. Anything that risks making our fork tests less reliable is a blocker i'm afraid.

Reply:
More constructively: Please make a self contained subtest (all new tests are required to use subtests), and make sure that it waits until the forked process is stopped safely.

Reply:
Not familiar with the `subtest` thing. `t/mojo/prefork.t` did not use these.
Are you suggesting that I put *all* the preforking tests into a single subtest?

With my little webapp, I can collect statistics about all the tests done and make sure everything is working as expected: that callbacks are invoked the expected amount of time and in the right processes.  Since we have many concurrent processes, I thought it would be nicer to cleanly abstract the statistics collection into a single tracking service that every sub-process knows how to reach and which has a single REST API.

I can add a `waitpid()` call at the end to make sure we cleanup properly, but I have added a failsafe in the webapp by instantiating a callback every 0.5 second to check whether the child has been re-parented to `init` (sorry, one should say `systemd` nowadays).  Did I do that right?

Reply:
Please leave the existing tests alone, and treat everything new as a separate new subtest. We will convert the rest of `prefork.t` to subtests in time. Make the test app no more complicated than absolutely necessary. Every line that looks unnecessary will be questioned and might end up as a blocker.

Reply:
The conversion is an ongoing process independent of this PR. https://github.com/mojolicious/mojo/issues/1520